### ISSUE TITLE
TypeError on ionic state reset

### ISSUE THREAD

**Issue by wouldgo** [driftyco/ionic-cli#422]
Hi,

just upgraded to the newest version of cli and now when I ran the `ionic state reset` command I see this issue:

```
TypeError: Cannot read property 'platforms' of undefined
    at Object.restoreState (/home/dario/.nvm/versions/node/v0.12.2/lib/node_modules/ionic/node_modules/ionic-app-lib/lib/state.js:405:15)
    at Object.resetState (/home/dario/.nvm/versions/node/v0.12.2/lib/node_modules/ionic/node_modules/ionic-app-lib/lib/state.js:747:9)
    at Object.run (/home/dario/.nvm/versions/node/v0.12.2/lib/node_modules/ionic/lib/ionic/state.js:773:13)
    at Object.run (/home/dario/.nvm/versions/node/v0.12.2/lib/node_modules/ionic/lib/cli.js:71:32)
    at Object.<anonymous> (/home/dario/.nvm/versions/node/v0.12.2/lib/node_modules/ionic/bin/ionic:12:10)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

Cannot read property 'platforms' of undefined (CLI v1.4.0)

Your system information:

Cordova CLI: 5.0.0
Gulp version:  CLI version 3.8.11
Gulp local:   Local version 3.8.11
Ionic Version: 1.0.0-rc.5
Ionic CLI Version: 1.4.0
Ionic App Lib Version: 0.0.16
OS: Distributor ID:	Ubuntu Description:	Ubuntu 14.04.2 LTS 
Node Version: v0.12.2
```

A friend of mine (with os x) has the same problem.

Investigating the stack error I don't undestand why [here](https://github.com/driftyco/ionic-cli/blob/1.4.0/lib/ionic/state.js#L773) you pass two parms, [here](https://github.com/driftyco/ionic-app-lib/blob/0.0.16/lib/state.js#L747) they're ignored but, lastly [here](https://github.com/driftyco/ionic-app-lib/blob/0.0.16/lib/state.js#L405) there's no checks against the possibility of an `undefined` value.

There is something that I miss?
Thanks a lot and best regards

Dario

**Comment by jbavari** [comment]
Nice catch! What you're seeing here is - refactoring mistakes out of one project into another. 

Patching it now. Thanks for reporting!

**Comment by wouldgo** [comment]
No problems! :smile: Good to have spotted for the community.

Thanks again

**Comment by jbavari** [comment]
Patched in 1.4.1 - enjoy!
